core/arch/interp.c

```diff
@@ -3377,13 +3381,15 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             bb->instr_start = bb->cur_pc;
             if (bb->full_decode) {
                 /* only going through this do loop once! */
-                bb->cur_pc = decode(dcontext, bb->cur_pc, bb->instr);
+                bb->cur_pc = IF_AARCH64_ELSE(decode_with_ldstex, decode)
```

Comment:
I can't think of a better name... decode_with_ldstex_blocks?

(One reason for not decoding first: it might be worth searching for LDXR/STXR with a bit mask to avoid the cost of decoding, or caching decoded instructions, because there's a performance problem here.)